### PR TITLE
fix: Improve CMP zipslip protection

### PR DESCRIPTION
**Pull request by leoluz**
Signed-off-by: Leonardo Luz Almeida <leonardo_almeida@intuit.com>

This PR implements stronger protection against zip-slip and symlink exploits in cmp-server

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: util/io/files/tar.go [hunk 1]
```diff
@@ -73,30 +72,30 @@ func Untgz(dstPath string, r io.Reader) error {
...
 		// Sanity check to protect against zip-slip
-		if !strings.HasPrefix(target, filepath.Clean(dstPath)+string(os.PathSeparator)) {
+		if !Inbound(target, dstPath) {
 			return fmt.Errorf("illegal filepath in archive: %s", target)
 		}
 
 		switch header.Typeflag {
 		case tar.TypeDir:
-			err := os.MkdirAll(target, 0755)
+			err := os.MkdirAll(target, 0755) // lgtm[go/zipslip]
```
Comment (alexmt) [inline util/io/files/tar.go:82]: As I understand code scanning does not really respect `// lgtm[go/zipslip]`, right? Please remove it if it is not used.

**Review by alexmt — APPROVED** [review]
LGTM.

Added one minor comment

**Review by crenshaw-dev — APPROVED** [review]
LGTM after Alex's comment. :-)

File: util/io/files/tar.go (hunk 1 above)
Comment (leoluz) [reply util/io/files/tar.go:82]: I don't understand why it doesn't. I opened an issue with codeql-go project to understand this better:
https://github.com/github/codeql-go/issues/703

Removing this for now.

**Pull request merged**